packages/library-base/src/commonMain/kotlin/io/realm/kotlin/exceptions/RealmException.kt

```diff
@@ -9,7 +9,7 @@ package io.realm.kotlin.exceptions
  */
 public open class RealmException : RuntimeException {
     public constructor() : super()
-    public constructor(message: String) : super(message)
-    public constructor(message: String, cause: Throwable) : super(message, cause)
-    public constructor(cause: Throwable) : super(cause)
+    public constructor(message: String?) : super(message)
```

Comment:
It was actually on purpose I made these non-null as we should always strive to have some kind of message. I can see you use `null` in the `safeErrorCodeMapping` method, but I would consider if we could just return `""` there instead since it is kind of an edge case.